plugin/pkg/scheduler/factory/factory.go

```diff
@@ -312,7 +312,58 @@ func (f *ConfigFactory) CreateFromKeys(predicateKeys, priorityKeys sets.String,
 		}
 	}
 
-	pluginArgs := PluginFactoryArgs{
+	predicateFuncs, err := f.GetPredicates(predicateKeys)
```

Comment:
Done. Slightly different than what you're proposing, but should meet your criteria :)